### DISCUSSION TITLE
Can I do call-graph analysis on top of tree sitter for python and java projects?

### DISCUSSION THREAD

**Discussion by smith-co** [tree-sitter/tree-sitter#2999] [Q&A] [answered]
I am curious to know whether is it possible to use tree-sitter to get all the caller/callee functions and also to get dependencies such as classes/packages. Is there any such tools already developed by anyone?

Anyone aware of any such implementation?

Please help.

**Comment by sogaiu** [comment, marked as answer]
May be the following are in the neighborhood of interest?

https://github.com/tree-sitter/tree-sitter-graph
https://dcreager.net/talks/stack-graphs/
